### ISSUE TITLE
Subdirectory attribute in config file isn't respected

### ISSUE THREAD

**Issue by brightgen-cni** [keirbowden/LDNSCall2020Plugin#9]
The "subdirectory" attribute in the config.json file seems to be ignored by the bbdoc:doc command.

I was trying to generate two versions of the DD and wanted to use objectsR1/objectsR2 and triggersR1/triggersR2. So I tried to use this attribute in config but it is always generating out puts in the "objects" and "triggers" subfolders. So I have to manually copy the generated outputs into the right sub-folder.

Not a big issue but thought it may be useful it it works as expected.

Thanks

**Comment by keirbowden** [comment]
The subdirectory attribute defines where the plug-in should look to find the metadata, not where the report output will be generated. It won't be a lot of work to make the output directory configurable, but the index.html will only link to the pages created by the final run. A longer term solution could be to allow multiple "objects" stanzas in the configuration file, but without knowing the details of what you are trying to achieve it's a bit of a guess.

I'll treat this issue as a feature request to make the report subdirectory for a metadata type configurable.

**Comment by keirbowden** [comment]
The report subdirectory for a metadata type is configurable in version 3.4.5 - blog post to follow.

**Comment by keirbowden** [comment]
I've also raised an issue to document the configuration file properties, so there is no confusion in the future.

**Comment by brightgen-cni** [comment]
Thank you so much, Keir.

Yeah I was a bit confused (and thanks for correcting). I hadn't realised the generated html files didn't actually have any metadata contents when I changed subdirectory attribute.

Your enhancement will allow me to maintain multiple DDs distributed across different work-stream branches, while also avoiding conflicts later when merging them. I have tweaked my index.html a bit (save the footer info of course) that allows me to publish these DDs on a single page of customer's portal.

Thanks again and best regards.
